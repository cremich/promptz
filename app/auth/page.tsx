"use client";

import Login from "@/components/Login";

export default function AuthPage() {
  return <div>{/* <Login /> */}</div>;
}
