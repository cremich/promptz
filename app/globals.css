pre,
code {
  word-wrap: break-word !important;
  white-space: pre-wrap !important;
  white-space: -moz-pre-wrap !important;
  white-space: -pre-wrap !important;
  white-space: -o-pre-wrap !important;
  margin: 0;
  padding: 0;
}

code {
  background-color: #f3f3f7;
}

pre > code {
  display: inline-block;
  padding: 1em;
}
