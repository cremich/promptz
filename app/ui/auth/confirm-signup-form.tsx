"use client";
import { Button } from "@/components/ui/button";
import {
  Card,
  CardContent,
  CardDescription,
  CardHeader,
  CardTitle,
} from "@/components/ui/card";
import { Input } from "@/components/ui/input";
import { Label } from "@/components/ui/label";
import { Mail } from "lucide-react";
import { ConfirmState, handleConfirmSignUp } from "@/app/lib/actions/cognito";
import { ErrorMessage } from "@/app/ui/error-message";
import { useActionState } from "react";

export function ConfirmSignUpForm() {
  const initialState: ConfirmState = {
    message: null,
    errors: {},
  };
  const [state, formAction] = useActionState(handleConfirmSignUp, initialState);

  return (
    <div className="flex flex-col gap-6">
      <Card>
        <CardHeader>
          <CardTitle className="text-2xl flex items-center">
            <Mail className="mr-2" />
            You&apos;ve got mail
          </CardTitle>
          <CardDescription>
            Your confirmation code is on the way. Enter the code we emailed you
            to confirm your account creation. It may take a minute to arrive.
          </CardDescription>
        </CardHeader>
        <CardContent>
          <form action={formAction}>
            <div className="flex flex-col gap-6">
              <div className="grid gap-2">
                <div className="flex items-center">
                  <Label htmlFor="code">Confirmation Code</Label>
                </div>
                <Input
                  id="code"
                  name="code"
                  type="text"
                  required
                  placeholder="123456"
                  maxLength={6}
                />
                {state.errors?.code && (
                  <ErrorMessage description={state.errors.code.join(" ")} />
                )}
              </div>
              {state.message && <ErrorMessage description={state.message} />}
              <Button type="submit" variant="default" className="w-full">
                Confirm
              </Button>
            </div>
          </form>
        </CardContent>
      </Card>
    </div>
  );
}
