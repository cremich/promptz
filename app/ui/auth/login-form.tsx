"use client";
import { Button } from "@/components/ui/button";
import {
  Card,
  CardContent,
  CardDescription,
  CardHeader,
  CardTitle,
} from "@/components/ui/card";
import { Input } from "@/components/ui/input";
import { Label } from "@/components/ui/label";
import Link from "next/link";
import {
  handleSignIn,
  handleSignUp,
  LoginState,
} from "@/app/lib/actions/cognito";
import { useState, useActionState } from "react";
import { ErrorMessage } from "@/app/ui/error-message";

export function LoginForm() {
  const initialState: LoginState = {
    message: null,
    errors: {},
  };
  const [email, setEmail] = useState("");
  const [state, formAction] = useActionState(handleSignIn, initialState);
  return (
    <div className={"flex flex-col gap-6"}>
      <Card>
        <CardHeader>
          <CardTitle className="text-2xl">Login</CardTitle>
          <CardDescription>
            Enter your email below to login to your account or login with your
            Google account.
          </CardDescription>
        </CardHeader>
        <CardContent>
          <form action={formAction}>
            <div className="flex flex-col gap-6">
              <div className="grid gap-2">
                <Label htmlFor="email">Email</Label>
                <Input
                  id="email"
                  type="email"
                  name="email"
                  placeholder="m@example.com"
                  required
                  value={email}
                  onChange={(e) => setEmail(e.target.value)}
                />
              </div>
              {state.errors?.email &&
                state.errors.email.map((error: string) => (
                  <ErrorMessage description={error} />
                ))}
              <div className="grid gap-2">
                <div className="flex items-center">
                  <Label htmlFor="password">Password</Label>
                  <a
                    href="#"
                    className="ml-auto inline-block text-sm underline-offset-4 hover:underline"
                  >
                    Forgot your password?
                  </a>
                </div>
                <Input id="password" type="password" name="password" required />
              </div>
              <Button type="submit" className="w-full">
                Login
              </Button>
              <hr></hr>
              <Button variant="outline" className="w-full">
                Login with Google
              </Button>
            </div>
            <div className="mt-4 text-center text-sm">
              Don&apos;t have an account?{" "}
              <Link
                key="log-in"
                href="/signup"
                className="underline underline-offset-4"
              >
                Create an account
              </Link>
            </div>
          </form>
        </CardContent>
      </Card>
    </div>
  );
}
