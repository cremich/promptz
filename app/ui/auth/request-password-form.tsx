"use client";
import { Button } from "@/components/ui/button";
import {
  Card,
  CardContent,
  CardDescription,
  CardHeader,
  CardTitle,
} from "@/components/ui/card";
import { Input } from "@/components/ui/input";
import { Label } from "@/components/ui/label";
import Link from "next/link";
import {
  handleRequestPassword,
  RequestPasswordState,
} from "@/app/lib/actions/cognito";
import { useState, useActionState } from "react";
import { ErrorMessage } from "@/app/ui/error-message";

export function RequestPasswordForm() {
  const initialState: RequestPasswordState = {
    message: null,
    errors: {},
  };
  const [email, setEmail] = useState("");
  const [state, formAction] = useActionState(
    handleRequestPassword,
    initialState,
  );
  return (
    <div className={"flex flex-col gap-6"}>
      <Card>
        <CardHeader>
          <CardTitle className="text-2xl">Forgot Your Password</CardTitle>
          <CardDescription>
            Enter your email below to reset your password.
          </CardDescription>
        </CardHeader>
        <CardContent>
          <form action={formAction}>
            <div className="flex flex-col gap-6">
              <div className="grid gap-2">
                <Label htmlFor="email">Email</Label>
                <Input
                  id="email"
                  type="email"
                  name="email"
                  placeholder="m@example.com"
                  required
                  value={email}
                  onChange={(e) => setEmail(e.target.value)}
                />
              </div>
              {state.errors?.email &&
                state.errors.email.map((error: string) => (
                  <ErrorMessage description={error} />
                ))}
              <Button type="submit" className="w-full">
                Request Password Reset
              </Button>
            </div>
            <div className="mt-4 text-center text-sm">
              Remember your password?{" "}
              <Link
                key="log-in"
                href="/login"
                className="underline underline-offset-4"
              >
                Log In
              </Link>
            </div>
          </form>
        </CardContent>
      </Card>
    </div>
  );
}
