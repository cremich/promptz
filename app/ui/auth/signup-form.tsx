"use client";
import { Button } from "@/components/ui/button";
import {
  Card,
  CardContent,
  CardDescription,
  CardHeader,
  CardTitle,
} from "@/components/ui/card";
import { Input } from "@/components/ui/input";
import { Label } from "@/components/ui/label";
import { useActionState, useState } from "react";
import { handleSignUp, SignUpState } from "@/app/lib/actions/cognito";
import Link from "next/link";
import { ErrorMessage } from "@/app/ui/error-message";

export function SignUpForm() {
  const initialState: SignUpState = {
    message: null,
    errors: {},
  };
  const [email, setEmail] = useState("");
  const [state, formAction] = useActionState(handleSignUp, initialState);

  return (
    <div className="flex flex-col gap-6">
      <Card>
        <CardHeader>
          <CardTitle className="text-2xl">Create Your Account</CardTitle>
          <CardDescription>
            Enter your email, and preferred username below to create a new
            account for Promptz.
          </CardDescription>
        </CardHeader>
        <CardContent>
          <form action={formAction} className="space-y-3">
            <div className="flex flex-col gap-6">
              <div className="grid gap-2">
                <Label htmlFor="email">Email</Label>
                <Input
                  id="email"
                  type="email"
                  name="email"
                  placeholder="m@example.com"
                  required
                  value={email}
                  onChange={(e) => setEmail(e.target.value)}
                />
              </div>
              {state.errors?.email &&
                state.errors.email.map((error: string, index: number) => (
                  <ErrorMessage description={error} key={index} />
                ))}
              <div className="grid gap-2">
                <div className="flex items-center">
                  <Label htmlFor="username">Username</Label>
                </div>
                <Input id="username" type="text" name="username" required />
              </div>
              {state.errors?.username &&
                state.errors.username.map((error: string, index: number) => (
                  <ErrorMessage description={error} key={index} />
                ))}
              <Button type="submit" className="w-full">
                Create Account
              </Button>
              {state.message && <ErrorMessage description={state.message} />}
            </div>
            <div className="mt-4 text-center text-sm">
              Already have an account?{" "}
              <Link
                key="log-in"
                href="/login"
                className="underline underline-offset-4"
              >
                Log In
              </Link>
            </div>
          </form>
        </CardContent>
      </Card>
    </div>
  );
}
