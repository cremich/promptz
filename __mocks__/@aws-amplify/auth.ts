import { jest } from "@jest/globals";

export const signOut = jest.fn();
export const signUp = jest.fn();
export const signIn = jest.fn().mockImplementation(() => {
  return Promise.resolve({
    nextStep: {
      signInStep: "DONE",
    },
  });
});
export const confirmSignUp = jest.fn().mockImplementation(() => {
  return Promise.resolve({
    isSignedIn: true,
    nextStep: {
      signUpStep: "COMPLETE_AUTO_SIGN_IN",
    },
  });
});
export const autoSignIn = jest.fn();
export const resetPassword = jest.fn();
export const confirmResetPassword = jest.fn();
