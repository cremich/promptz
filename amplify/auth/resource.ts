import { defineAuth, secret } from "@aws-amplify/backend";
import { verificationEmailTemplate } from "./email-templates";

const getSenders = () => {
  return process.env["PROMPTZ_ENV"] === "sandbox"
    ? undefined
    : {
        email: {
          fromEmail: "noreply@promptz.dev",
        },
      };
};

const getExternalProviders = () => {
  return process.env["PROMPTZ_ENV"] === "sandbox"
    ? undefined
    : {
        google: {
          clientId: secret("GOOGLE_CLIENT_ID"),
          clientSecret: secret("GOOGLE_CLIENT_SECRET"),
          attributeMapping: {
            email: "email",
            emailVerified: "email_verified",
            preferredUsername: "name",
          },
          scopes: ["email", "openid", "profile"],
        },
        callbackUrls: [
          "https://promptz.dev/auth",
          "https://www.promptz.dev/auth",
        ],
        logoutUrls: ["https://promptz.dev/", "https://www.promptz.dev"],
      };
};

/**
 * Define and configure your auth resource
 * @see https://docs.amplify.aws/gen2/build-a-backend/auth
 */
export const auth = defineAuth({
  loginWith: {
    email: {
      verificationEmailStyle: "CODE",
      verificationEmailSubject: "Welcome to PROMPTZ!",
      verificationEmailBody: (createCode) =>
        verificationEmailTemplate(createCode),
    },
    externalProviders: getExternalProviders(),
  },
  senders: getSenders(),
  userAttributes: {
    preferredUsername: {
      required: true,
    },
  },
});
