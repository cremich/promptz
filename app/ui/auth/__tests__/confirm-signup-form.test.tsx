import { describe, expect, test } from "@jest/globals";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import { ConfirmSignUpForm } from "@/app/ui/auth/confirm-signup-form";

describe("Confirm Signup Form", () => {
  test("renders confirm signup form unchanged", () => {
    const { container } = render(<ConfirmSignUpForm />);
    expect(container).toMatchSnapshot();
  });
});
