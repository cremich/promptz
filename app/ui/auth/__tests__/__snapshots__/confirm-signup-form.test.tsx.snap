// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirm Signup Form renders confirm signup form unchanged 1`] = `
<div>
  <div
    class="flex flex-col gap-6"
  >
    <div
      class="rounded-lg border bg-neutral-800 text-card-foreground shadow-sm"
    >
      <div
        class="flex flex-col space-y-1.5 p-6"
      >
        <div
          class="font-semibold tracking-tight text-2xl flex items-center"
        >
          <svg
            class="lucide lucide-mail mr-2"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              height="16"
              rx="2"
              width="20"
              x="2"
              y="4"
            />
            <path
              d="m22 7-8.97 5.7a1.94 1.94 0 0 1-2.06 0L2 7"
            />
          </svg>
          You've got mail
        </div>
        <div
          class="text-sm text-muted-foreground"
        >
          Your confirmation code is on the way. Enter the code we emailed you to confirm your account creation. It may take a minute to arrive.
        </div>
      </div>
      <div
        class="p-6 pt-0"
      >
        <form
          action="javascript:throw new Error('A React form was unexpectedly submitted. If you called form.submit() manually, consider using form.requestSubmit() instead. If you\\'re trying to use event.stopPropagation() in a submit event handler, consider also calling event.preventDefault().')"
        >
          <div
            class="flex flex-col gap-6"
          >
            <div
              class="grid gap-2"
            >
              <div
                class="flex items-center"
              >
                <label
                  class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
                  for="code"
                >
                  Confirmation Code
                </label>
              </div>
              <input
                class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-base ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 md:text-sm"
                id="code"
                maxlength="6"
                name="code"
                placeholder="123456"
                required=""
                type="text"
              />
            </div>
            <button
              class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 bg-primary text-primary-foreground shadow hover:bg-primary/90 h-9 px-4 py-2 w-full"
              type="submit"
            >
              Confirm
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
